lib/src/client.dart

```diff
@@ -470,6 +491,16 @@ class ClientImpl implements Client {
     request.name = _config.name;
     request.version = _config.version;
 
+    if (isWeb()) {
+      // Use headers emulation in web context.
+      final Map<String, String> emulatedHeaders = Map.fromEntries(
+        _headers.entries
+            .where((entry) => entry.value is String)
+            .map((entry) => MapEntry(entry.key, entry.value as String)),
+      );
+      request.headers.addAll(emulatedHeaders);
+    }
```

Comment:
```dart
if (kIsWeb) {
   // It would be a dead code and tree shaked at Dart VM (dart:io) platforms
   // Use headers emulation in web context.
   final emulatedHeaders = <String, Object?>{
     for (final MapEntry(:String key, :Object? value) in _headers.entries)
        if (value is String) key: value
        else if (value is Iterable) key: value.whereType<String>().join(', '),
  };
}
```

Reply:
That work this way:

```dart
void main() {
  final headers = {
    '1': 'a',
    '2': ['b', 'c', 'd'],
    '3': 'e',
  };
  print(<String, String>{
     for (final MapEntry(:String key, :Object? value) in headers.entries)
        if (value is String) key: value
        else if (value case Iterable iterable)
          for (final e in iterable.whereType<String>())
            key: e,
  }); // {1: a, 2: d, 3: e}
}
```

Reply:
Thx, not actual anymore